### PR TITLE
updatecli: enable dependabot, refactor and use signed commits

### PR DESCRIPTION
**Pull request by v1v**
### What

* Enable dependabot for GiHub actions
* Support signed commits
* Move from `.buildkite` to `.github/updatecli.d`, it uses GitHub actions.
* Use .github/actions for the reusable GitHub composite actions.

### Why

Tidy up a bit the project and use verified commits in case someone in the future decide to use that particular branch protection.

Similar to https://github.com/elastic/beats/pull/39472

### CODE REVIEW

File: .github/workflows/bump-golang-previous.yml [hunk 1]
```diff
@@ -23,3 +23,4 @@ jobs:
           vaultUrl: ${{ secrets.VAULT_ADDR }}
           vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
           vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
+          command: '--experimental apply'
```
Comment (amannocci) [inline .github/workflows/bump-golang-previous.yml:26]: This input doesn't seems to be present in the `./.github/actions/bump-golang` workflow.

File: .github/workflows/bump-golang.yml [hunk 2]
```diff
@@ -13,12 +13,13 @@ jobs:
...
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/bump-golang
         with:
           branch: 'main'
           # NOTE: when a new golang version please update me with 1.<go-version>
           go-minor: '1.22'
           vaultUrl: ${{ secrets.VAULT_ADDR }}
           vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
           vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
+          command: '--experimental apply'
```
Comment (amannocci) [inline .github/workflows/bump-golang.yml:25]: This input doesn't seems to be present in the `./.github/actions/bump-golang` workflow.

Comment (v1v) [reply .github/workflows/bump-golang.yml:25]: good catch!

https://github.com/elastic/golang-crossbuild/pull/403/commits/812759ab6be0eaa05620da9126aef2d4459b5c81

File: .github/workflows/bump-golang-previous.yml (hunk 1 above)
Comment (v1v) [reply .github/workflows/bump-golang-previous.yml:26]: [`812759a` (#403)](https://github.com/elastic/golang-crossbuild/pull/403/commits/812759ab6be0eaa05620da9126aef2d4459b5c81)

File: .github/actions/bump-golang/action.yml [hunk 3]
```diff
@@ -24,6 +24,10 @@ inputs:
   vaultSecretId:
     description: 'Vault secret ID'
     required: true
+  command:
+    description: 'What updatecli command'
+    default: 'apply'
```
Comment (amannocci) [inline .github/actions/bump-golang/action.yml:29]: The new input isn't used in the composite steps.

**Review by amannocci — APPROVED** [review]

**Comment by v1v** [comment]
> @v1v some of the GH actions are still looking for .buildkite and are not broken: [elastic/golang-crossbuild/actions/runs/9073091796/job/24929651799#step:3:330](https://github.com/elastic/golang-crossbuild/actions/runs/9073091796/job/24929651799#step:3:330)

I missed to backport this, I'm working on the fix, thanks for reaching out :)

**Comment by v1v** [comment]
@mergifyio backport 1.21

**Pull request merged**